zospy/zpcore.py

```diff
@@ -374,15 +376,15 @@ def _assign_connection(self):
         else:
             logger.debug("ZOSAPI_Connection() already assigned self.Connection")
 
-    def connect_as_extension(self, instancenumber: int = 0) -> bool:
-        """Creates a standalone Zemax Opticstudio instance.
+    def connect_as_extension(self, instancenumber: int = 0) -> OpticStudioSystem:
```

Comment:
```suggestion
    def connect_as_extension(self, instancenumber: int = 0, return_system: bool = False) -> bool | OpticStudioSystem:
```

I have discussed this with @crnh and @jwmbeenakker. We are in favor of directly returning the opticstudiosystem, but fear that it might be a breaking change for some users. Therefore, we would like to implement it as a boolean argument for now. This way, the change is not a breaking change for now. When we approach a version that will induce breaking changes, we can change the default behaviour.

Please update the docsting and function return accordingly

Reply:
@crnh do we want this argument to be named `return_system` or `return_opticstudiosystem`, or differently?